### PR TITLE
Format code with clang-format

### LINKED ISSUE #286: Style guide: #include sorting
This is about the style guide.

Right now there does not seem to be any fixed ordering/sorting of includes. Furthermore, sometimes relative includes are used (`#include "../dialogs/projectpropertieseditordialog.h"`) and sometimes not (`#include <librepcb/common/dialogs/boarddesignrulesdialog.h>`).

Sorting of includes has the advantage that pull requests adding includes conflict less often. Is there a de-facto standard for include ordering? Maybe something that could be included in the QtCreator style file?

Otherwise I'd suggest the following ordering:

- 1: C++ imports (`#include <iostream>`)
- 2: Qt imports (`#include <QtCore>`)
- 3: Imports from other modules, always starting with `librepcb` (`#include <librepcb/project/project.h>`)
- 4: Imports relative to the current module (`#include "fsm/bes_fsm.h"`)

All imports should be sorted alphanumerically inside their group.

@ubruhin @rnestler

### PR DESCRIPTION
**Pull request by ubruhin**
- [x] Add `.clang-format`
- [x] Add bash script to run `clang-format` on all required files
- [x] Update QtCreator coding style configuration and `.editorconfig`
- [x] Update documentation (e.g. README.md, CONTRIBUTING.md or Doxygen documentation)
- [x] Format code
- [x] Make CI reporting failure if files with Windows line endings or tabulators are checked in

Fixes #286.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Auto
+TabWidth:        8
+UseTab:          Never
+...
+
```
Comment (dbrgn) [inline .clang-format]: Oh, now we need a reformatter for the `.clang-format` file :stuck_out_tongue_closed_eyes:

Comment (ubruhin) [reply .clang-format]: Hmm why? The file was generated by `clang-format` itself...

Comment (dbrgn) [reply .clang-format]: Sorry, that was a bad recursive joke...

**Comment by ubruhin** [comment]
Who wants to review? :grin:

File: dev/format_code.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+echo "Formatting modified files with clang-format..."
+
+COUNTER=0
+
+for dir in apps/ libs/librepcb/ tests/unittests/
+do
+  MODIFIED=`git diff --name-only master -- "${dir}**.cpp" "${dir}**.hpp" "${dir}**.h"`
+  UNTRACKED=`git ls-files --others --exclude-standard -- "${dir}**.cpp" "${dir}**.hpp" "${dir}**.h"`
+  for file in $MODIFIED $UNTRACKED
```
Comment (dbrgn) [inline dev/format_code.sh]: That's an elegant solution!

File: dev/format_code.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  do
+    # Note: Do NOT use in-place edition of clang-format because this causes
+    # "make" to detect the files as changed every time, even if the content was
```
Comment (rnestler) [inline dev/format_code.sh]: That sounds like a horrible bug in `clang-format` which makes it almost unusable :confused:

**Pull request merged**